### PR TITLE
git_util - accept `git.Repo` path 

### PR DESCRIPTION
**Pull request by adi88d**
## Related Issues
fixes:
https://jira-dc.paloaltonetworks.com/browse/XSUP-38518

## Description
Updated GitUtil constructor to accept `git.Repo` in `path` parameter.

### CODE REVIEW

File: demisto_sdk/commands/common/git_util.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def __init__(
 
         if isinstance(path, str):
             repo_path = Path(path)
+        elif isinstance(path, self.REPO_CLS):
```
Comment (dorschw) [inline demisto_sdk/commands/common/git_util.py:58]: Update the type declaration in line 52 to allow `Repo` as well as the existing `str, Path`.

Comment (adi88d) [reply demisto_sdk/commands/common/git_util.py:58]: done

File: demisto_sdk/commands/common/tests/git_util_test.py [hunk 2]
```diff
@@ -49,3 +49,22 @@ class Object:
...
+
+    Then
+        - Ensure the GitUtil repo path is not None.
+    """
+    from demisto_sdk.commands.common.git_util import GitUtil
+
+    repo = GitUtil.REPO_CLS()
+
+    git_util = GitUtil(repo)
+    assert git_util.repo is not None
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/git_util_test.py:70]:
```suggestion
    assert git_util.repo is not None
    assert git_util.repo_path == repo.working_dir
```

Comment (adi88d) [reply demisto_sdk/commands/common/tests/git_util_test.py:70]: git_util is not contains repo_path, changed to:
 `git_util.repo.working_dir`

File: demisto_sdk/commands/common/tests/git_util_test.py [hunk 3]
```diff
@@ -49,3 +49,23 @@ class Object:
...
+    Then
+        - Ensure the GitUtil repo path equals to the repo path.
+    """
+    from demisto_sdk.commands.common.git_util import GitUtil
+
+    repo = GitUtil.REPO_CLS()
+
+    git_util = GitUtil(repo)
+    assert git_util.repo is not None
+    assert git_util.repo.working_dir == repo.working_dir
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/git_util_test.py:71]:
```suggestion
    assert git_util.repo == repo
```

**Review by dorschw — APPROVED** [review]

**Pull request merged**